src/dataframe/dataframe.jl

```diff
@@ -512,6 +517,10 @@ function Base.setindex!(df::DataFrame, val::Any, col_inds::AbstractVector{<:Colu
     return df
 end
 
+# df[Regex] = value
```

Comment:
The existing code says "Single Item" rather than "value".

Reply:
This is exactly the point - I have written `value` on purpose. This method is catch-all for any `v` and dispatches accordingly.

Using this approach we "inherit" what is allowed from other `setindex!` definitions. When we remove implicit broadcasting in the other methods it will automatically disallowed for `Regex` specification of columns.

Reply:
I do the same in line https://github.com/JuliaData/DataFrames.jl/pull/1819/files#diff-0aa1ff52d56badabf7d1511d3dc3f279R684 for `setindex!` where row range is passed.